applications/serial_lte_modem/doc/slm_description.rst

```diff
@@ -0,0 +1,818 @@
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSOCKET=0**
+         #XSOCKET: 0, closed
+         OK
+
+#. Test a connection-based UDP client with UDP proxy service (note that these commands are available only if :option:`CONFIG_SLM_UDP_PROXY` is defined):
+
+   a. Check the available values for the XUDPCLI command.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPCLI=?**
+         #XUDPCLI: (0, 1, 2),<url>,<port>,<sec_tag>
+         OK
+
+   #. Create a UDP client and connect to a server.
+      Replace *example.com* with the host name or IPv4 address of a UDP server and *1234* with the corresponding port.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPCLI=1,"**\ *example.com*\ **",**\ *1234*
+         #XUDPCLI: 2 connected
+         OK
+
+   #. Send plain text data to the UDP server and check the returned data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPSEND=1,"Test UDP"**
+         #XUDPSEND: 8
+         OK
+         #XUDPRECV: 1, 14
+         PONG: Test UDP
+
+   #. Send hexadecimal data to the UDP server and check the returned data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPSEND=0,"DEADBEEF"**
+         #XUDPSEND: 4
+         OK
+         #XUDPRECV: 0, 20
+         504F4E473A20DEADBEEF
+
+   #. Disconnect from the server.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPCLI=0**
+         OK
+
+#. Test a connection-based UDP client with UDP proxy service in data mode (note that these commands are available only if :option:`CONFIG_SLM_UDP_PROXY` is defined):
+
+   a. Create a UDP client and connect to a server with data mode support.
+      Replace *example.com* with the host name or IPv4 address of a UDP server and *1234* with the corresponding port.
+      Then read information about the connection.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPCLI=2,"**\ *example.com*\ **",**\ *1234*
+         #XUDPCLI: 1 connected
+         OK
+
+         **AT#XUDPCLI?**
+         #XUDPCLI: 1, 1
+         OK
+
+   #. Send plain text data to the UDP server and retrieve the returned data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **Test UDP by hostname**
+         PONG: Test UDP by hostname
+
+   #. Disconnect from the server.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPCLI=0**
+         #XUDPCLI: disconnected
+         OK
+
+
+TLS client
+~~~~~~~~~~
+
+Before completing this test, you must update the CA certificate, the client certificate, and the private key to be used for the TLS connection in the modem.
+The credentials must use the security tag 16842755.
+
+To store the credentials in the modem, use LTE Link Monitor.
+See `Managing credentials`_ in the LTE Link Monitor User Guide for instructions.
+
+You must register the corresponding credentials on the server side.
```

Comment:
Is that correct?

Reply:
It depends on whether server need to authenticate the client.
TLS/DTLS client, no need
TLS server/HTTPS/MQTT over TLS, @lats1980